docs/src/pages/components/data-grid/selection/DisableRowSelection.js

```diff
@@ -1,31 +1,19 @@
 import * as React from 'react';
 import { DataGrid } from '@material-ui/data-grid';
-import { randomUserName, randomInt } from '@material-ui/x-grid-data-generator';
-
-const columns = [
-  { field: 'id' },
-  { field: 'username', width: 150 },
-  { field: 'age', width: 80, type: 'number' },
-];
-
-const rows = [
-  { id: 1, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 2, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 3, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 4, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 5, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 6, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 7, username: randomUserName(), age: randomInt(10, 80) },
-  { id: 8, username: randomUserName(), age: randomInt(10, 80) },
-];
+import { useDemoData } from '@material-ui/x-grid-data-generator';
 
 export default function DisableRowSelection() {
+  const { data } = useDemoData({
+    dataSet: 'Commodity',
+    rowLength: 100,
+    maxColumns: 6,
+  });
+
   return (
     <div style={{ height: 400, width: '100%' }}>
       <DataGrid
-        columns={columns}
-        rows={rows}
-        isRowSelectable={(params) => Number(params.id) % 2 === 0}
+        {...data}
+        isRowSelectable={(params) => params.row.quantity > 50000}
```

Comment:
why not the status filled as you suggested?

Reply:
The user would have to scroll to see the column.

Reply:
Ah yeah I saw in the demo page.
As a doc showcase, I think you should go very basic, no need to use the commodity IMHO

Reply:
Let's keep the commodity, it's closer to a real-world application than odd/even rows. AG-Grid has a similar [example](https://ag-grid.com/react-grid/row-selection/#specify-selectable-rows) disabling rows when some property is below a value.

Reply:
My assumption was that using the commodity dataset would make the demo source easier to follow (because less code and more consistent) and the demo itself easier to follow because more consistent with the other demos.

Reply:
It seems a bit overkill to me for inline demos...

Reply:
I guess that an inlined dataset, if we use the same one between different demos would work as well.

My assumption is that we should be very cautious to not make developers learn the structure of the data rendered for each demo, but keep them short and consistent.